feat(input): initial design guidance (#28)
vmware-clarity/clarity-guidance: 1014-input-design-guidance.md

---
category: component design guidance
state: draft
created: 2024-10-15
updated: 2024-10-22
---

# Input Design Guidance

> The [input](https://clarity.design/documentation/input) component enables the user to input a single sentence or less of text.

Reviewer: ```suggestion
The input component allows users to enter a single sentence or less of text, providing a straightforward way to capture concise information.
```

## Guidance

- **Must** use an input when you need to allow user to enter a single sentence or less of text.

> - **Must not** use an input in case of multiple lines of text. Use a [textarea](https://clarity.design/documentation/textarea) for multiple lines of text input.

Reviewer: ```suggestion
-Design teams **must not** use an input for capturing multiple lines of text. Instead, use a [textarea](https://clarity.design/documentation/textarea) for this purpose.
```

> - **Should** provide a clear and concise visible label for context.

Reviewer: ```suggestion
- Design teams **should** provide a clear and concise visible label for context, helping users understand what information is expected in the input field.
```

> - If there is no visible label, an `aria-label` (or other accessible name) **must** be provided and **should** be called out in the design.

Reviewer: ```suggestion
- Design teams **must** provide an `aria-label` (or other accessible name) if there is no visible label, ensuring the input is accessible to all users. This accessible name **should** be clearly documented in the design to promote consistency and understanding.
```

> - **Should not** use placeholders as a substitute for a proper label element. It **may** be used to show an example of the type of data that should be entered into a form. 

Reviewer: ```suggestion
- Design teams **must not** use placeholders as a substitute for a proper label element, as this creates accessibility challenges for users. Placeholders can be invisible when typing and increase cognitive load, leading to confusion about what information is required.  Placeholders **may** be used to provide examples of the type of data that should be entered into a form, but they should never replace clear and descriptive labels.
```

> - **Should not** use placeholders in place of helper text.

Reviewer: ```suggestion
- Design teams **must not** use placeholders in place of helper text, as this can lead to confusion about the expected input. Helper text **should** provide additional guidance and context, which placeholders cannot effectively communicate.
```

---
category: component design guidance
state: draft
created: 2024-10-15
updated: 2024-10-21
---

# Input Design Guidance

> [Input](https://clarity.design/documentation/input) component enable the user to input a single sentence or less of text.

Reviewer 2: ```suggestion
The [input](https://clarity.design/documentation/input) component enables the user to input a single sentence or less of text.
```

## Guidance

> - **Must** use Input when you need to allow user to enter a single sentence or less of text.

Reviewer 2: ```suggestion
- **Must** use an input when you need to allow user to enter a single sentence or less of text.
```

> - **Must** NOT use Input in case of multiple lines of text. Use [textarea](https://clarity.design/documentation/textarea) component to support multiple lines of text input.

Reviewer 2: ```suggestion
- **Must not** use an input in case of multiple lines of text. Use a [textarea](https://clarity.design/documentation/textarea) for multiple lines of text input.
```

> - **Should** provide a label with a clear and concise text for context. 

Reviewer 2: ```suggestion
- **Should** provide a clear and concise visible label for context.
- If there is no visible label, an `aria-label` (or other accessible name) **must** be provided and **should** be called out in the design.
```

I think this should be clarified for accessibility.

I made this same comment for the textarea guidance: https://github.com/vmware-clarity/clarity-guidance/pull/34#discussion_r1808969101

- **Should** NOT use placeholders as a substitute for a proper label element. It may only be used to show an example of the type of data that should be entered into a form. 

> - **Should** NOT use placeholders in place of helper text.

Reviewer 2: Please sync this with the input guidance. https://github.com/vmware-clarity/clarity-guidance/pull/34/files#r1808903933
